DNN Platform/Library/Entities/Users/UserRegistrationEmailNotifier.cs

```diff
@@ -35,7 +38,9 @@ public static void NotifyUser(UserInfo user)
             switch (PortalSettings.Current.UserRegistration)
             {
                 case (int)PortalRegistrationType.PrivateRegistration:
-                    NotifyUser(user, MessageType.UserRegistrationPrivate);
+                    NotifyUser(user, CurrentUser.Value != null && CurrentUser.Value.IsSuperUser ?
```

Comment:
I don't think `IsSuperUser` is the right check here.  This should apply for _any_ user created by an admin (or, really, anyone with permission to create users), right?  In the case of private registration, isn't that everyone?  Our goal is to avoid sending this message:
>Your account details will be reviewed by the website Administrator and you will receive a notification upon account activation.

Is that message _ever_ true when using private registration?  If registration is private, users cannot register themselves, and so there is no review.

If there needs to be any check here, maybe it should be a check for whether the user is approved or in the unverified users role.

Reply:
@daguiler since you originally reported the issue, am I missing anything? ☝️ 

It seems like we just need to change that message, rather than check anything.

Reply:
@bdukes , I'm afraid you are wrong. 

1. Private registration means we require administrators to approve user accounts. So even in a Private mode anyone is able to register himself. 

![image](https://user-images.githubusercontent.com/22524011/59453935-b136dc80-8e19-11e9-86d2-ad5666901592.png)

2. Another thing is that we notify users on a post user creation even `OnUserCreated`. That means user is already created and db entry exists at the moment we send email. In such case, do we really need to check admin roles and other permission staff? Look like it is senseless. I even think `IsSuperUser` check is not needed here. If someone is logged, that means registration happened by someone with user creation right. If nobody logged, registration comes from outside, so we require approval. Am I right? It's enough to check if `CurrentUser` is not null.

https://github.com/dnnsoftware/Dnn.Platform/blob/1a867de6f65f1d5472eed5ca25dfef5cd504a77a/DNN%20Platform/Library/Entities/Users/UserEventHandlers.cs#L37-L45

cc: @sleupold

Reply:
Ah, you're right, I was thinking of `NoRegistration`.

I think your check for whether someone is logged in is simple enough, but you need to check whether `UserID != Null.NullInteger` because `GetCurrentUserInfo` returns an empty `UserInfo` instead of `null` if they aren't logged in.
https://github.com/dnnsoftware/Dnn.Platform/blob/1a867de6f65f1d5472eed5ca25dfef5cd504a77a/DNN%20Platform/Library/Entities/Users/UserController.cs#L209-L225

Reply:
You are right. Just tested it. Works as expected now!